metadata/src/main/java/com/redhat/lightblue/metadata/ValueGenerator.java

```diff
@@ -0,0 +1,42 @@
+package com.redhat.lightblue.metadata;
+
+import java.io.Serializable;
+import java.util.Properties;
+
+/**
+ * Represents value generator definition located in entityInfo.
+ *
+ * @author mpatercz
+ *
+ */
+public class ValueGenerator implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    public enum ValueGeneratorType {
+        IntSequance;
+    }
+
```

Comment:
Types are value types, not field types. You use Type to convert a value
to/from Json. We'll put the generated value to a Json document, so Type is
the correct way of doing it, you simply call type.toJson(generatedValue)

On Tue, Aug 18, 2015 at 3:04 AM, Marek <notifications@github.com> wrote:

> In
> metadata/src/main/java/com/redhat/lightblue/metadata/ValueGenerator.java
> <https://github.com/lightblue-platform/lightblue-core/pull/431#discussion_r37277909>
> :
>
> > +import java.util.Properties;
> > +
> > +/**
> > + * Represents value generator definition located in entityInfo.
> > + *
> > + * @author mpatercz
> > + *
> > + */
> > +public class ValueGenerator implements Serializable {
> > +
> > +    private static final long serialVersionUID = 1L;
> > +
> > +    public enum ValueGeneratorType {
> > +        IntSequance;
> > +    }
> > +
>
> From what I see, com.redhat.lightblue.metadata.types are field types.
> Generator is not a field. com.redhat.lightblue.metadata.types.Type provides
> methods like to/fromJson, supportsEq, supportsOrdering - how (and why)
> would those be used for generator types?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/431/files#r37277909>
> .
>